numpy/core/src/multiarray/descriptor.c

```diff
@@ -280,15 +280,21 @@ _convert_from_tuple(PyObject *obj, int align)
                     "invalid shape in fixed-type tuple.");
             goto fail;
         }
-        /*
-         * If (type, 1) was given, it is equivalent to type...
-         * or (type, ()) was given it is equivalent to type...
-         */
-        if ((shape.len == 1
-                    && shape.ptr[0] == 1
-                    && PyNumber_Check(val))
-                || (shape.len == 0
-                    && PyTuple_Check(val))) {
+        /* if (type, ()) was given it is equivalent to type... */
+        if (shape.len == 0 && PyTuple_Check(val)) {
+            npy_free_cache_dim_obj(shape);
+            return type;
+        }
+        /* (type, 1) use to be equivalent to type, but is deprecated */
+        if (shape.len == 1
+                && shape.ptr[0] == 1
+                && PyNumber_Check(val)) {
+            if (DEPRECATE_FUTUREWARNING(
```

Comment:
Deprecation warnings need a comment giving the version and date they were introduced. Search the source for use of the deprecate macro and you should see and example

Reply:
done